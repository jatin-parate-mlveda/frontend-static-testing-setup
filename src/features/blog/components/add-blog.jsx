import { useState } from 'react';
import { ReUsableButton } from 'shared';

export const AddBlog = () => {
  const [title, setTitle] = useState('');
  const [desc, setDesc] = useState('');

  return (
    <form
      onSubmit={e => {
        e.preventDefault();
      }}
    >
      <div>
        <label htmlFor='title'>
          Title
          <input
            name='title'
            id='title'
            value={title}
            onChange={e => setTitle(e.target.value)}
          />
        </label>
      </div>
      <div>
        <label htmlFor='desc'>
          Description
          <input
            name='desc'
            id='desc'
            value={desc}
            onChange={e => setDesc(e.target.value)}
          />
        </label>
      </div>
      <div>
        <ReUsableButton isSubmitBtn text='Submit' type='submit' />
      </div>
    </form>
  );
};
