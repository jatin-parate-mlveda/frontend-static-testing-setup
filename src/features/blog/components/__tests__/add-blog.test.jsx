import { render } from '@testing-library/react';
import fireEvent from '@testing-library/user-event';
import { AddBlog } from '../add-blog';

test('matches snapshot', () => {
  const { container } = render(<AddBlog />);

  expect(container.outerHTML).toMatchSnapshot();
});

test('submits form correctly', () => {
  const onBlogAdd = jest.fn();

  const { getByLabelText } = render(<AddBlog onAdd={onBlogAdd} />);
  const title = 'title'; // TODO: use faker
  const desc = 'description'; // TODO: use faker

  const titleInput = getByLabelText('Title');
  const descInput = getByLabelText('Description');
  const { form } = titleInput;

  fireEvent.type(titleInput, title);
  fireEvent.type(descInput, desc);
  form.submit();

  expect(onBlogAdd).toHaveBeenCalledTimes(1);
  expect(onBlogAdd).toBeCalledWith({ title, desc });
});
