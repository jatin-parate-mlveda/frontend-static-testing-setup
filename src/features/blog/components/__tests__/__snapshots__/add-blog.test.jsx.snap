// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `"<div><form><div><label for=\\"title\\">Title<input name=\\"title\\" id=\\"title\\" value=\\"\\"></label></div><div><label for=\\"desc\\">Description<input name=\\"desc\\" id=\\"desc\\" value=\\"\\"></label></div><div><button type=\\"submit\\" class=\\" button\\">Submit</button></div></form></div>"`;
